Initial direct type wrappers (for c++)

rubidge opened the pull request:
## Motivation and Context
* Provide some c++ type wrappers corresponding to Types, that can be directly used (not via Python).

## Approach
* Provide some c++ type wrappers corresponding to Types:
    str for String
    ListOf<> for ListOfType()
    TupleOf<> for TupleOfType()
* Rudimentary generation of NamedTuple wrappers
    examples: NamedTupleBoolAndInt, NamedTupleBoolIntStr, NamedTupleListAndTupleOfStr
* seeking code review
* Also, finished implementing String.split(), using existing methods to build list.

## How Has This Been Tested?
* String.split() has test cases
* new wrapper types not tested (verify approach first)

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

braxtonmckee commented:
Sweet! I will take a look and revert 

Sent from my iPhone

> On Apr 2, 2019, at 7:31 PM, rubidge <notifications@github.com> wrote:
> 
> Motivation and Context
> 
> Provide some c++ type wrappers corresponding to Types, that can be directly used (not via Python).
> Approach
> 
> Provide some c++ type wrappers corresponding to Types:
> str for String
> ListOf<> for ListOfType()
> TupleOf<> for TupleOfType()
> Rudimentary generation of NamedTuple wrappers
> examples: NamedTupleBoolAndInt, NamedTupleBoolIntStr, NamedTupleListAndTupleOfStr
> seeking code review
> Also, finished implementing String.split(), using existing methods to build list.
> How Has This Been Tested?
> 
> String.split() has test cases
> new wrapper types not tested (verify approach first)
> Types of changes
> 
>  Bug fix (non-breaking change which fixes an issue)
>  New feature (non-breaking change which adds functionality)
>  Breaking change (fix or feature that would cause existing functionality to not work as expected)
> Checklist:
> 
>  My code follows the code style of this project.
>  My change requires a change to the documentation.
>  I have updated the documentation accordingly.
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/braxtonmckee/nativepython/pull/76
> 
> Commit Summary
> 
> Working on direct type wrappers for use in c++
> Forgot important file in last commit
> Fix types_test for ListOf
> Lint corrections
> Add rudimentary direct NamedTuple generator
> File Changes
> 
> A gentype.py (29)
> M nativepython/tests/string_compilation_test.py (84)
> M nativepython/type_wrappers/runtime_functions.py (24)
> M nativepython/type_wrappers/string_wrapper.py (52)
> M typed_python/AllTypes.hpp (1)
> M typed_python/ConstDictType.cpp (2)
> M typed_python/ConstDictType.hpp (4)
> A typed_python/DirectTypes.hpp (439)
> M typed_python/PyTupleOrListOfInstance.cpp (24)
> M typed_python/PyTupleOrListOfInstance.hpp (22)
> M typed_python/PythonSerializationContext.cpp (8)
> M typed_python/StringType.cpp (147)
> M typed_python/StringType.hpp (5)
> M typed_python/TupleOrListOfType.cpp (53)
> M typed_python/TupleOrListOfType.hpp (20)
> M typed_python/Type.hpp (8)
> M typed_python/_runtime.cpp (17)
> M typed_python/_types.cpp (10)
> M typed_python/types_test.py (2)
> Patch Links:
> 
> https://github.com/braxtonmckee/nativepython/pull/76.patch
> https://github.com/braxtonmckee/nativepython/pull/76.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

rubidge commented:
Braxton:
Thanks, hope this is a step in the right direction.
-Jeff

On Tue, Apr 2, 2019 at 7:37 PM Braxton McKee <notifications@github.com>
wrote:

> Sweet! I will take a look and revert
>
> Sent from my iPhone
>
> > On Apr 2, 2019, at 7:31 PM, rubidge <notifications@github.com> wrote:
> >
> > Motivation and Context
> >
> > Provide some c++ type wrappers corresponding to Types, that can be
> directly used (not via Python).
> > Approach
> >
> > Provide some c++ type wrappers corresponding to Types:
> > str for String
> > ListOf<> for ListOfType()
> > TupleOf<> for TupleOfType()
> > Rudimentary generation of NamedTuple wrappers
> > examples: NamedTupleBoolAndInt, NamedTupleBoolIntStr,
> NamedTupleListAndTupleOfStr
> > seeking code review
> > Also, finished implementing String.split(), using existing methods to
> build list.
> > How Has This Been Tested?
> >
> > String.split() has test cases
> > new wrapper types not tested (verify approach first)
> > Types of changes
> >
> > Bug fix (non-breaking change which fixes an issue)
> > New feature (non-breaking change which adds functionality)
> > Breaking change (fix or feature that would cause existing functionality
> to not work as expected)
> > Checklist:
> >
> > My code follows the code style of this project.
> > My change requires a change to the documentation.
> > I have updated the documentation accordingly.
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/braxtonmckee/nativepython/pull/76
> >
> > Commit Summary
> >
> > Working on direct type wrappers for use in c++
> > Forgot important file in last commit
> > Fix types_test for ListOf
> > Lint corrections
> > Add rudimentary direct NamedTuple generator
> > File Changes
> >
> > A gentype.py (29)
> > M nativepython/tests/string_compilation_test.py (84)
> > M nativepython/type_wrappers/runtime_functions.py (24)
> > M nativepython/type_wrappers/string_wrapper.py (52)
> > M typed_python/AllTypes.hpp (1)
> > M typed_python/ConstDictType.cpp (2)
> > M typed_python/ConstDictType.hpp (4)
> > A typed_python/DirectTypes.hpp (439)
> > M typed_python/PyTupleOrListOfInstance.cpp (24)
> > M typed_python/PyTupleOrListOfInstance.hpp (22)
> > M typed_python/PythonSerializationContext.cpp (8)
> > M typed_python/StringType.cpp (147)
> > M typed_python/StringType.hpp (5)
> > M typed_python/TupleOrListOfType.cpp (53)
> > M typed_python/TupleOrListOfType.hpp (20)
> > M typed_python/Type.hpp (8)
> > M typed_python/_runtime.cpp (17)
> > M typed_python/_types.cpp (10)
> > M typed_python/types_test.py (2)
> > Patch Links:
> >
> > https://github.com/braxtonmckee/nativepython/pull/76.patch
> > https://github.com/braxtonmckee/nativepython/pull/76.diff
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/braxtonmckee/nativepython/pull/76#issuecomment-479255828>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/At14i9su2GIAzbtpaIFeDzVDQyNACowoks5vc-nJgaJpZM4cZUNq>
> .
>

rubidge commented:
superseded
